### ISSUE TITLE
test数据集的使用

### ISSUE THREAD

**Issue by yuhangT** [HKUST-KnowComp/BMGF-RoBERTa#1]
请问为何要在训练过程保存在test数据集上最优的模型，这样划分验证集和test集的意义是不是就没有了？这样得到的结果和别的模型作比较是否不公平？

**Comment by seanliu96** [comment]
Hyperparameter tuning was conducted and selected over the validation data. And all experiments and ablation studies were under the same setting.

**Comment by thelinhbkhn2014** [comment]
When I conducted experiments to reproduce your results based on your command lines that you provided, it seems that you reported results through hyperparameter tuning on the test data. This should not happen in research.
P/s: although if you use results through hyperparameter tuning on the validation set, the results just reduced about 1-2% compared to your results on your paper. 
Anw, can you provide your command lines for train_binary.py file on the PDTB2.0 dataset?

**Comment by seanliu96** [comment]
Hi, thanks for your interest in our work. Our reported models were saved based on the evaluation over validation data. We ran multiple times and averaged the accuracies/f1 scores. If you cannot reproduce our experiments, please try our seeds 0, 1000, 1996, 2000, 2020. It's worth noting that we might get different results even when we used the same seeds. If you have the same problem, please set "save_best" as "accf1" because it makes the difference between validation and test performance as close as possible.

For the commands for "train_binary.py", the only change is to explicit set "relations" as "Comparison"/"Contingency"/"Expansion"/"Temporal".

**Comment by thelinhbkhn2014** [comment]
Thank you for your reply. I tried to run your code for the binary classification, but it had a bug in Sampler class. When I fixed this bug in the same way you fixed it in the train.py file, there was a bug in the evaluation. Can you re-run your code to check it?

**Comment by seanliu96** [comment]
Updated. Please have a try.
